### PR TITLE
Flathub

### PR DESCRIPTION
**Pull request by danigm**
This PR adds a new option to the generate_manifest script to generate a manifest to publish on flathub easily.

https://phabricator.endlessm.com/T30976

### CODE REVIEW

File: com.hack_computer.Fizzics/data/metainfo.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <component type="desktop">
   <id>com.hack_computer.Fizzics.desktop</id>
   <metadata_license>CC0</metadata_license>
-  <project_license>LicenseRef-proprietary</project_license>
+  <project_license>GPL-2.0</project_license>
```
Comment (manuq) [inline com.hack_computer.Fizzics/data/metainfo.xml:5]: :+1:

File: generate_manifest [hunk 2]
```diff
@@ -52,6 +60,13 @@ fi
 APP_ID=$1
 FLATPAK_BRANCH=${FLATPAK_BRANCH:-master}
 GIT_BRANCH=${GIT_BRANCH:-master}
+GIT_PATH=${GIT_PATH:-.}
+GIT_COMMIT=""
+if [ -n "$FLATHUB" ]; then
+  # If flathub we get the branch commit sha to add to the manifest
+  COMMIT=$(wget -O - https://api.github.com/repos/endlessm/hack-toy-apps/git/ref/heads/${GIT_BRANCH} 2> /dev/null | grep sha | cut -d '"' -f4)
```
Comment (manuq) [inline generate_manifest:67]: Have you considered getting the SHA with git directly?
```
$ git fetch && git rev-parse origin/master
6c6c141aa96a469928da3b393f454ff3abf081af
```
I'm not sure about introducing a github specific api call. Which one you think is better?

File: manifest.json.in [hunk 3]
```diff
@@ -15,18 +15,11 @@
   "command": "@APP_ID@",
   "finish-args": [
     "--share=ipc",
-    "--socket=x11",
+    "--socket=fallback-x11",
```
Comment (manuq) [inline manifest.json.in:18]: :+1:

File: manifest.json.in [hunk 4]
```diff
@@ -15,18 +15,11 @@
...
     "--socket=pulseaudio",
     "--device=dri",
     "--own-name=@APP_ID@",
-    "--filesystem=xdg-run/dconf",
-    "--filesystem=~/.config/dconf:ro",
-    "--nofilesystem=~/.local/lib/python3.5",
-    "--talk-name=ca.desrt.dconf",
-    "--filesystem=host",
-    "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
-    "--env=GSETTINGS_SCHEMA_DIR=/run/host/usr/share/glib-2.0/schemas/",
```
Comment (manuq) [inline manifest.json.in:29]: :+1:

File: meson.build [hunk 5]
```diff
@@ -39,6 +39,11 @@ install_data(
     install_dir: 'share/icons/hicolor/64x64/apps',
     rename: app_id + '.png'
 )
+install_data(
+    join_paths(app_id, 'data', 'app.icon.png'),
+    install_dir: 'share/icons/hicolor/128x128/apps',
+    rename: app_id + '.png'
```
Comment (manuq) [inline meson.build:45]: :+1:

**Review by manuq — APPROVED** [review]
Looks great. I left an alternative approach for obtaining the commit SHA. Feel free to take it.

File: generate_manifest (hunk 2 above)
Comment (danigm) [reply generate_manifest:67]: Yes, with git should be easier, I'll change that. It's just I didn't find the correct way to do that and sometimes I don't see the simplest solution and try to do something more complex. But your solution is a lot simpler.

**Pull request merged**